.travis.yml

```
@@ -20,23 +20,17 @@ jobs:
     - stage: "Testing stage"
       name: "Tests (all modules) + Coverage"
       script:
-        - go get -u github.com/mattn/goveralls
-        - go get -u golang.org/x/tools/cmd/cover
         - make deps
-        - go test -v -coverprofile=coverage.out -covermode=count ./...
-        - $HOME/gopath/bin/goveralls -coverprofile=coverage.out -service=travis-ci
+        - go test -v -race -coverprofile=coverage.txt -covermode=atomic ./...
```

@hsanjuan travis is going to keep on failing because running `go test ./...` doesn't output anything to stdout until all the tests have completed, which with `-race` is longer than the travis timeout. So we need to run `go test` on each directory individually so that it outputs along the way.

```suggestion
        - go test -v -race -coverprofile=coverage.txt -covermode=atomic ./...
        +echo 'mode: atomic' > coverage.txt
        +for d in `find -name '*test.go' -printf '%h\n' | sort -u`
        +do
        +    go test -v -race -coverprofile=coverage-$d.txt -covermode=atomic .
        +    tail -n+2 coverage-$d.txt >> coverage.txt
        +    rm coverage-$d.txt
        +done
```
First we create the header of the coverage.txt file, then we find all the directories that have go test files in them and then for each dir we find we run `go test -race` and outputting a unique coverage report file. The coverage-$d.txt file is piped into the aggregate file minus the `mode: atomic` header and then the directory specific coverage file is removed.

---

We just need to prepend `travis_wait`: https://docs.travis-ci.com/user/common-build-problems/#build-times-out-because-no-output-was-received